src/Controller/ResettingController.php

```diff
@@ -0,0 +1,166 @@
+use DateTime;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Csrf\TokenGenerator\TokenGeneratorInterface;
+
+class ResettingController extends AbstractController
+{
+    /**
+     * @Route("/request", name="request_resetting")
+     * @param Request $request
+     * @param EntityManagerInterface $entityManager
+     * @param TokenGeneratorInterface $tokenGenerator
+     * @param MailerInterface $mailer
+     * @return RedirectResponse|Response
+     * @throws TransportExceptionInterface
+     */
+    public function request(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        TokenGeneratorInterface $tokenGenerator,
+        MailerInterface $mailer
+    ) {
+        //création formulaire
+
+        $user = $this->getUser();
+
+        $form = $this->createFormBuilder()
+            ->add('email', EmailType::class)
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $email = $form->getData();
+            $entityManager = $this->getDoctrine()->getManager();
+
+            $user = $entityManager->getRepository(User::class)
+                ->findOneBy([
+                    'email' => $email
+                ]);
+
+
+            if (!$user) {
+                $this->addFlash('danger', "Cet email n'existe pas");
+                return $this->redirectToRoute("request_resetting", [
+                    'form' => $form->createView()
+                ]);
+            } else {
+                $user->setToken($tokenGenerator->generateToken());
+                $user->setPasswordRequestedAt(new DateTime());
+
+                $entityManager->persist($user);
+
+                $entityManager->flush();
+
+
+                $mailUser = (new Email())
+                    ->from(new Address('github-test@bipbip-mobile.fr', 'BipBip Mobile'))
+                    ->to(new Address($user->getEmail()))
+                    ->subject('Réinitialisation du mot de passe')
+                    ->html($this->renderView('security/passLink.html.twig', array(
+                        'user' => $user
+                    )));
+
+                $request->getSession()->getFlashBag()->add('success', "Un mail va vous être envoyé afin que
```

Comment:
Cette formulation fonctionne-t-elle ? D'habitude on utilise `$this->addFlash('success', "Message");`

Reply:
oui ça fonctionne, c'est le seul qui fonctionne